### PR TITLE
Fix: systemd services

### PR DESCRIPTION
**Pull request by Owen-sz**

### CODE REVIEW

**Comment by madonuko** [comment]
why is there a 404?

**Comment by Owen-sz** [comment]
> why is there a 404?

Upstream had v in their versions, fixed now in this PR.

File: anda/system/surface-dtx-daemon/surface-dtx-daemon.spec [hunk 1]
```diff
@@ -48,6 +48,19 @@ install -D -m644 "target/_surface-dtx-userd" "%{buildroot}/usr/share/zsh/site-fu
 install -D -m644 "target/surface-dtx-daemon.fish" "%{buildroot}/usr/share/fish/vendor_completions.d/surface-dtx-daemon.fish"
 install -D -m644 "target/surface-dtx-userd.fish" "%{buildroot}/usr/share/fish/vendor_completions.d/surface-dtx-userd.fish"
 
+# These systemd services should be included in the preset file for Ultramarine Linux Surface images
+%post
+%systemd_post surface-dtx-daemon.service
+%systemd_post surface-dtx-userd.service
```
Comment (madonuko) [inline anda/system/surface-dtx-daemon/surface-dtx-daemon.spec]: this is a user service, use %systemd_user_post instead

File: anda/system/cros-keyboard-map/cros-keyboard-map.spec [hunk 2]
```diff
@@ -41,14 +41,18 @@ mkdir -p %buildroot%{_unitdir}
...
 
 %preun
-%systemd_preun 88-ultramarine-chromebook-default.preset
+%systemd_preun cros-keyboard-map.service
+%systemd_preun keyd.service
 
 %postun
-%systemd_postun_with_restart 88-ultramarine-chromebook-default.preset
+%systemd_postun_with_restart cros-keyboard-map.service
+%systemd_postun_with_restart keyd.service
```
Comment (madonuko) [inline anda/system/cros-keyboard-map/cros-keyboard-map.spec]: this service does not exist

**Review by madonuko — CHANGES_REQUESTED** [review]
please also bump releases

File: anda/system/cros-keyboard-map/cros-keyboard-map.spec (hunk 2 above)
Comment (madonuko) [reply anda/system/cros-keyboard-map/cros-keyboard-map.spec]: keyd.service is already handled in the keyd package, there is no reason to cross reference keyd here

Comment (Owen-sz) [reply anda/system/cros-keyboard-map/cros-keyboard-map.spec]: Makes sense

**Comment by Owen-sz** [comment]
Releases are all up to date already

**Comment by madonuko** [comment]
> Releases are all up to date already

please increment the value for the `Release:` preamble

File: anda/system/surface-dtx-daemon/surface-dtx-daemon.spec [hunk 3]
```diff
@@ -48,6 +49,19 @@ install -D -m644 "target/_surface-dtx-userd" "%{buildroot}/usr/share/zsh/site-fu
...
+%systemd_post surface-dtx-daemon.service
+%systemd_user_post surface-dtx-userd.service
+
+%preun
+%systemd_preun surface-dtx-daemon.service
+%systemd_user_preun surface-dtx-userd.service
+
+%postun
+%systemd_postun_with_restart surface-dtx-daemon.service
+%systemd_user_postun surface-dtx-userd.service
```
Comment (madonuko) [inline anda/system/surface-dtx-daemon/surface-dtx-daemon.spec]:
```suggestion
%systemd_user_postun_with_restart surface-dtx-userd.service
```
please be careful next time

File: anda/system/surface-dtx-daemon/surface-dtx-daemon.spec [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
+%global ver2 %(echo %{ver} | sed 's/^v//')
 
 Name:           surface-dtx-daemon
 Version:        %(echo %ver | sed 's/-/~/g')
 Release:        1%{?dist}
 Summary:        Surface Detachment System (DTX) Daemon
 License:        MIT
 URL:            https://github.com/linux-surface/surface-dtx-daemon
-Source:         %url/archive/refs/tags/v%ver.tar.gz
+Source:         %url/archive/refs/tags/%ver.tar.gz
```
Comment (madonuko) [inline anda/system/surface-dtx-daemon/surface-dtx-daemon.spec:11]: if the tag does not start with `v`, then the auto update script would not add a `v`. This contradicts with the definition `%global ver v0.3.8-1`.

Comment (madonuko) [reply anda/system/surface-dtx-daemon/surface-dtx-daemon.spec:11]: if there is a new version / tag to `linux-surface/surface-dtx-daemon`, please bump `ver` manually.

also, you might no longer need `%ver2` depending on the situation

**Review by madonuko — CHANGES_REQUESTED** [review]

File: anda/system/surface-dtx-daemon/surface-dtx-daemon.spec (hunk 4 above)
Comment (Owen-sz) [reply anda/system/surface-dtx-daemon/surface-dtx-daemon.spec:11]: Well yeah, the surface-dtx-daemon releases do have the v in them, so the url it was trying to download had 2 'v's in it.
If I really need the v before %ver in the source url, manual bumping is fine, but I'd rather not?

**Review by madonuko — APPROVED** [review]

**Comment by raboneko** [comment]
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|f39|Backport failed because of merge conflicts|
|❌|f40|Backport failed because of merge conflicts|
|✅|f41|[](https://github.com/terrapkg/packages/pull/2759)|
|❌|el10|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
backport --pr 2752
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport) and see the [Github Action logs](https://github.com/terrapkg/packages/actions/runs/12513380614) for details

**Pull request merged**